src/Vision/ImageAnnotatorTrait.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Vision;
+
+use google\cloud\vision\v1\FaceAnnotation;
+use google\cloud\vision\v1\Image;
+use google\cloud\vision\v1\ImageSource;
+use google\cloud\vision\v1\Position;
+use InvalidArgumentException;
+
+trait ImageAnnotatorTrait
+{
+    private static $URL_SCHEMES = [
+        'http',
+        'https',
+        'gs',
+    ];
+
+    private $client;
+
+    protected function setClient($client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * @param resource|string|Image $image
```

Comment:
This is the protobuf object `\google\cloud\vision\v1\Image`, not the handwritten `Image` class. It is probably clearer to fully qualify the name